### ISSUE TITLE
ngbDropDown [autoClose]="false" is not working on outside click

### ISSUE THREAD

**Issue by rammohangowd** [valor-software/ng2-bootstrap#5094]
Hi, autoClose=false is working fine for inside click. But not working for outside click. when i click outside if i set [autoClose]="false" it's only close when i click on the element i open it from and it wont close if click anywhere else outside. if i set [autoClose]="true" it closes if i click anywhere inside the dropdown and outside the dropdown. Please suggest me when this fix will be provided.

**Comment by ludmilanesvitiy** [comment]
@rammohangowd Hi! Looks like you don't need to override autoClose param. By default, the dropdown will be closed by outside click. 
If you still need something other, please, specify what use case you have and behaviour do you expect.

**Comment by rammohangowd** [comment]
@ludmilanesvitiy , Thank you so much for your response. But in my code default behavior is not working. when I click on the body content of the dropdown its closing. when I want to type something on the filter box on click itself its closing. so I have made [autoClose]="false", so that now its not closing. But outside click also not closing. So it's only close when i click on the element i open it from:( My code looks like below.

   ```
<div ngbDropdown class="d-inline-block" [autoClose]="false">
        <button id="dropdownBasic2" ngbDropdownToggle>{{row.length}} selected {{secoundryLabel}}</button>
        <div ngbDropdownMenu aria-labelledby="dropdownBasic2" (click)="$event.stopPropagation()">
          <div class="ngx-tree">
            <div class="ngx-tree-left col-6">
               <!-- <ngx-treeview [items]="items" [itemTemplate]="itemTemplate" [config]="config" -->
              <ngx-treeview [items]="items | ngxTreeview:'name'" [itemTemplate]="itemTemplate" [config]="config"
             
                (selectedChange)="onSelectedChange($event)">
              </ngx-treeview>
            </div>
            <div class="ngx-tree-right col-6 ">
              <p>Selections</p>
              <ul>
                <li *ngFor="let i of row">{{i.text}}
                  <img src="assets/images/icons/closeX_disabled.png" />
                </li>
              </ul>
              <span class="clearAll"><img src="assets/images/icons/delete-normal.png" /></span>
            </div>
          </div>
          <div class="btn-group ">
            <div class="col pr-0">
              <label>Selection is limited maximum of 5 segments</label>
              <label>Any changes will apply here will reset the changes, if any on the building filter</label>
            </div>
            <div class="col-3 d-flex pr-0 pl-0">
              <div class="discard" (click)="discardFilter()"></div>
              <div class="apply ml-1"></div>
            </div>
          </div>
        </div>
      </div>
```

**Comment by ludmilanesvitiy** [comment]
@rammohangowd Hi! For investigations your issue, could you, please, create a small reproduction on Plunker/StackBlitz ?

**Comment by rammohangowd** [comment]
@ludmilanesvitiy , I do not know how to create in plunker, but uploaded in github. below is the link. please download and npm install. Thank you so much for looking into it.
https://github.com/rammohangowd/ngxtree-dropdownclose

**Comment by ludmilanesvitiy** [comment]
@rammohangowd Thank you for creating reproduction at github. Now I see, that you are not using ngx-bootstrap library, you use https://www.npmjs.com/package/@ng-bootstrap/ng-bootstrap - need to ask your question to that community.
But I've made a bit investigation and could propose you to use ngx-bootstrap collapse for your goal. The result could be the next:
![Peek 2019-03-25 12-06](https://user-images.githubusercontent.com/18209841/54911273-c543ff80-4ef6-11e9-9940-9771d516cd97.gif)
I've created also example how to do it: https://github.com/rammohangowd/ngxtree-dropdownclose/pull/1/files
